### PR TITLE
Daniel/calendar

### PR DESCRIPTION
**Pull request by DanielGovrin**

### CODE REVIEW

File: src/components/calendar/datepickerstyles.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+.react-datepicker__current-month,
+.react-datepicker__day-name {
+  color: white;
+}
+
+.react-datepicker__day--outside-month {
+  visibility: hidden;
+}
```
Comment (TheAlmightyCrumb) [inline src/components/calendar/datepickerstyles.css:22]: What are these selectors? Don't they provide a better way to access and edit those styles?

File: src/_codux/boards/calendar-date/calendar-date.board.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export default createBoard({
+    name: 'CalendarDate',
+    Board: () => (
+        <ComponentWrapper settings={{}}>
+            <ContentSlot>
+                <div className={styles.container}>
+                    <CalendarDate isAvailable={true} date={9} isSelected={false} />
+                    <CalendarDate isAvailable={true} date={9} isSelected={true} />
+                    <CalendarDate isAvailable={false} date={9} isSelected={false} />
+                    <CalendarDate isAvailable={false} date={9} isSelected={true} />
```
Comment (idango10) [inline src/_codux/boards/calendar-date/calendar-date.board.tsx:15]: How about making `isAvailable` and `isSelected` optional, and then you can do something like this:
```js
<CalendarDate date={9} isAvailable />
<CalendarDate date={9} isAvailable isSelected />
<CalendarDate date={9} />
<CalendarDate date={9} isSelected />
```

Looks cleaner IMO.

File: src/_codux/boards/calendar-date/calendar-date.board.module.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.container {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 100vh;
```
Comment (idango10) [inline src/_codux/boards/calendar-date/calendar-date.board.module.scss:5]: Saw that in the other board you were using % for the width, doesn't really matter which one you choose here but I would either change this to % or the other one to viewport units for consistency.
Also if you choose viewport units, consider using `dvh` and `dvw`.

File: src/components/calendar-date/calendar-date.module.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    cursor: pointer;
+    border-radius: 50%;
+
+}
+
+.date {
+    color: white;
```
Comment (idango10) [inline src/components/calendar-date/calendar-date.module.scss:13]: Should use css vars.

https://github.com/codux-demos/bookings-starter/blob/main/src/styles/theme.module.scss

File: src/components/calendar/calendar.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        const lessonDate = lesson.slot?.endDate
+            ? format(parseISO(lesson.slot.endDate), "dd/MM/yyyy")
+            : "Invalid date";
+        if (lessonDate in accumulator) {
+            accumulator[lessonDate].push(lesson);
+        } else {
+            accumulator[lessonDate] = [lesson];
+        }
+        return accumulator;
+    }, {});
```
Comment (idango10) [inline src/components/calendar/calendar.tsx:30]: Can extract this to a function outside of `Calendar`.

**Review by idango10 — COMMENTED** [review]
Looks good overall, I'd add some kind of indication that you can't choose a specific date because that's pretty confusing as is.

https://github.com/codux-demos/bookings-starter/assets/75563024/a5fb8723-dd24-4e1e-a049-8a5bfe72eee0

File: src/components/calendar/datepickerstyles.css (hunk 1 above)
Comment (DanielGovrin) [reply src/components/calendar/datepickerstyles.css:22]: these selectors are the onlky way I found to change certain styles in the calendar. will look for other ways.

Comment (TheAlmightyCrumb) [reply src/components/calendar/datepickerstyles.css:22]:
```
// https://reactdatepicker.com/#example-custom-calendar-class-name
// from the Sources tab in devtools: example.scss
118. .rasta-stripes {
119.   .react-datepicker__week:nth-child(3n + 1) {
120.     background-color: #215005;
121.   }
122.   .react-datepicker__week:nth-child(3n + 2) {
123.     background-color: #eea429;
124.   }
125.   .react-datepicker__week:nth-child(3n + 3) {
126.     background-color: #a82a15;
127.   }
128. }
```
Shit... 🌚

**Pull request closed without merging**